pkg/config/remote/service/service.go

```diff
@@ -907,6 +911,40 @@ func filterNeededTargetFiles(neededConfigs []string, cachedTargetFiles []*pbgo.T
 	return filteredList, nil
 }
 
+func (s *CoreAgentService) apiKeyUpdateCallback() func(string, any, any) {
+	return func(setting string, _, newvalue any) {
+		if setting != "api_key" {
+			return
+		}
+
+		newKey, ok := newvalue.(string)
+		s.Lock()
+		defer s.Unlock()
+
+		if !ok {
+			log.Errorf("Could not convert API key to string")
+			return
+		}
```

Comment:
Same here as before, no need to acquire the lock if this happens, so let's move it before the lock.